### PR TITLE
Add args for all Multisig transaction types, and other

### PR DESCRIPTION
**Pull request by hansl**
Moving the account module tests to using automock which simplifies
them, and moving roles to being an enum instead of generic strings.

This is overall better QoL for the code.

### CODE REVIEW

File: src/many/Cargo.toml [hunk 1]
```diff
@@ -47,6 +47,8 @@ sha3 = "0.9.1"
 simple_asn1 = "0.5.4"
 signature = "1.3"
 static_assertions = "1.1.0"
+strum = "0.24.0"
```
Comment (fmorency) [inline src/many/Cargo.toml:50]: Is this used anywhere?

Comment (hansl) [reply src/many/Cargo.toml:50]: https://github.com/liftedinit/many-rs/pull/95/files#diff-a308dae0adf7823b4efd6fe570f06aaed4f246521bcbb4b15ffd2b5f1c2438c4R19-R21

Comment (hansl) [reply src/many/Cargo.toml:50]: I think you might have missed a file.

Comment (fmorency) [reply src/many/Cargo.toml:50]: I did miss a file. Still lgtm!

**Pull request merged**